### PR TITLE
Add new pipeline to test the subscription upgrade to the v1alpha2 version

### PR DESCRIPTION
**Pull request by VladislavPaskar**
Add the new upgrade pipeline to test from Kyma main with **v1alpha1** Subscription to the Kyma PR-version with the **v1alpha2** Subscription as per https://github.com/kyma-project/kyma/issues/15815

see the last green runs [here](https://status.build.kyma-project.io/?job=vladislavpaskar_test_of_prowjob_pre-main-kyma-gardener-gcp-subs)

### CODE REVIEW

File: prow/scripts/cluster-integration/helpers/eventing.sh [hunk 1]
```diff
@@ -264,3 +267,20 @@ function eventing::run_copy_crds() {
 
     log::success "Eventing copy-crds make target completed"
 }
+
+function eventing::deployV1Alpha2Subscription() {
```
[the remark this answers is not in the document]
Comment (VladislavPaskar) [reply prow/scripts/cluster-integration/helpers/eventing.sh]: renamed it

File: prow/scripts/cluster-integration/helpers/eventing.sh [hunk 2]
```diff
@@ -264,3 +267,20 @@ function eventing::run_copy_crds() {
...
+    log::info "Copying the CRDs to installation/eventing"
+    export ENABLE_NEW_CRD_VERSION="true"
+
+    pushd /home/prow/go/src/github.com/kyma-project/kyma/components/eventing-controller
+    make copy-crds
```
[the remark this answers is not in the document]
Comment (VladislavPaskar) [reply prow/scripts/cluster-integration/helpers/eventing.sh]: done

File: prow/scripts/cluster-integration/kyma-integration-gardener-subscription-upgrade.sh [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+function postUpgrade() {
+  # upgrade the kyma to the current PR state
+  export KYMA_SOURCE="PR-${PULL_NUMBER}"
+
+  # uses previously set KYMA_SOURCE
+  log::info "### Upgrading Kyma to $KYMA_SOURCE"
+  gardener::deploy_kyma --source "${KYMA_SOURCE}"
+
+  # upgrade to the new v1alpha2 Subscription version
+  eventing::deployV1Alpha2Subscription
```
[the remark this answers is not in the document]
Comment (VladislavPaskar) [reply prow/scripts/cluster-integration/kyma-integration-gardener-subscription-upgrade.sh]: Good idea, I refactored the script

**Review by mfaizanse — DISMISSED** [review]

**Review by dekiel — APPROVED** [review]

**Pull request merged**